hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto

```diff
@@ -867,10 +867,10 @@ message DeletedKeys {
 }
 
 message DeleteKeysResponse {
-    repeated KeyInfo deletedKeys = 1;
-    repeated KeyInfo unDeletedKeys = 2;
```

Comment:
I don't see any usage of it, and also I see no real usage of this result in the client. 
And this is a new API adding, and will be released in next ozone release.

One more reason, as we fail whole batch, I don't see real use case for it as of current behavior.

Reply:
HDDS-3916 is in progress to use it on client side to resume delete failures.

Reply:
Will restore the original behavior.

Reply:
Thanks. Makes sense to me.

Reply:
Added tests.

Reply:
bq. One more reason, as we fail whole batch, I don't see real use case for it as of current behavior.

I think the writebatch should contain only those that can be deleted. Those undeleted will be returned to client.

Reply:
The current behavior is we fail the entire batch if we are not able to delete the key.

So, proposal is to change this behavior? 

And also I see we fail to delete only when key not found. In this case, there is no real point in returning them as unDeletedKeys.

2nd case we fail is when checking ACLS,  when user does not have permission, in this case even if we return unDeletedKeys, not sure how it will be used.

3rd case, when reading DB failure, then also whole batch will be failed. As anyway, at that time OM will be terminated, as this is a critical error to avoid DB divergence.

So, still, I am not really sure what is real use case of returning unDeletedKeys and how that will be used. Just trying to understand it. 

The current PR, kept the behavior of existing behavior, where it used to fail entire batch.

Reply:
bq. The current behavior is we fail the entire batch if we are not able to delete the key. So, proposal is to change this behavior?

The goal is to not fail the whole deletion when something failed in the middle and return the list of deleted and undeleted to the client. We can't get there if delete and undelete list are removed from the proto. Here is the reason for keeping them for HDDS-3916. 

1. It can be expensive to lookup or retry the whole list again if one of the deletion failed. 
2. Returning undelete file list reduce the amount of unnecessary processing.

Also holding the bucket lock once for all may not be a good idea to large delete as that will prevent other clients operation on the same bucket. grab and release individual or in small batches give other clients opportunities to access buckets.

Reply:
Got it. So, when this information is returned if the plan is to retry, I will add them.

The reason for my question is, we fail when we don't have permission/key not found we return them, but if this information is useful, will add that back.

Reply:
As in HA, and when ratis is enabled as default for non-HA, it is a single thread executor now. So, this optimization is not really required. For non-HA, it might help, I will open a new Jira for this improvement.

Reply:
Created https://issues.apache.org/jira/browse/HDDS-3948 for this.

Reply:
Looks good to me.